### PR TITLE
access lists support

### PR DESCRIPTION
**Pull request by dorothy-zbornak**
## Description

- adds access list (EIP 2930) support,  which caused lots of side effects:
  - upgraded `ethereumjs-util` across the entire repo
  - upgraded `@ethereumjs/tx` (replaces `ethereumjs-tx`) in `@0x/subproviders` which meant a lot of subproviders had to get fixed
  - I think trezor was broken and might be fixed now but who knows
- also fixes a bug where JSON RPC responses with null `result` fields are considered errors.

### TODO
- [x] test generating access lists
- [x] test eth_call with access list
- [x] test eth_sendTransaction with access list ([tx](https://ropsten.etherscan.io/tx/0x6a0d8354623d747d10bb613fd7b6e745befd18790d2521d64a6f36d4ec0bce13#accesslist))

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: utils/src/abi_encoder/utils/math.ts [hunk 1]
```diff
@@ -19,10 +19,7 @@ function sanityCheckBigNumberRange(
     }
 }
 function bigNumberToPaddedBuffer(value: BigNumber): Buffer {
-    const valueHex = `0x${value.toString(constants.HEX_BASE)}`;
-    const valueBuf = ethUtil.toBuffer(valueHex);
-    const valueBufPadded = ethUtil.setLengthLeft(valueBuf, constants.EVM_WORD_WIDTH_IN_BYTES);
-    return valueBufPadded;
+    return ethUtil.setLengthLeft(ethUtil.toBuffer(new ethUtil.BN(value.toFixed(0))), constants.EVM_WORD_WIDTH_IN_BYTES);
```
Comment (dorothy-zbornak) [inline utils/src/abi_encoder/utils/math.ts:22]: It's like 13x faster on my machine but it doesn't seem like this is the bottleneck in encoding.

File: subproviders/src/subproviders/trezor.ts [hunk 2]
```diff
@@ -83,34 +91,28 @@ export class TrezorSubprovider extends BaseWalletSubprovider {
...
-            tx.raw[vIndex] = Buffer.from([1]); // v
-            const rIndex = 7;
-            tx.raw[rIndex] = Buffer.from([]); // r
-            const sIndex = 8;
-            tx.raw[sIndex] = Buffer.from([]); // s
-
-            // slice off leading 0x
-            tx.v = Buffer.from(payload.v.slice(2), 'hex');
-            tx.r = Buffer.from(payload.r.slice(2), 'hex');
-            tx.s = Buffer.from(payload.s.slice(2), 'hex');
```
Comment (dorothy-zbornak) [inline subproviders/src/subproviders/trezor.ts:111]: I feel like this accomplished nothing and maybe I fixed it

**Review by xianny — APPROVED** [review]
I didn't scrutinise everything but the parts I understand look good to me!

**Review by mzhu25 — APPROVED** [review]
thanks for updating all this stuff!

File: json-schemas/schemas/call_data_schema.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
                 "properties": {
                     "code": {
                         "type": "string",
-                        "pattern": "^0x[0-9a-fA-F]*$"
+                        "pattern": "^0x[0-9a-fA-F]{64}$"
```
[the remark this answers is not in the document]
Comment (dorothy-zbornak) [reply json-schemas/schemas/call_data_schema.json:30]: Doh, not bytes32

**Pull request merged**